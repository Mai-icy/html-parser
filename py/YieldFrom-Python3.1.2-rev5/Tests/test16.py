#
#   Test delegating 'throw' to non-generator
#

def g():
	try:
		print("Starting g")
		yield from range(10)
	finally:
		print("Finishing g")

gi = g()
for i in range(5):
	x = next(gi)
	print("Yielded", x)
e = ValueError("tomato ejected")
gi.throw(e)
