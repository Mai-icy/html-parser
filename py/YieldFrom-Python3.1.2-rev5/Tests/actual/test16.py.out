Starting g
Yielded 0
Yielded 1
Yielded 2
Yielded 3
Yielded 4
Finishing g
Traceback (most recent call last):
  File "test16.py", line 17, in <module>
    gi.throw(e)
  File "test16.py", line 8, in g
    yield from range(10)
ValueError: tomato ejected
